### PR TITLE
Merge anaconda-recipes back to master

### PR DESCRIPTION
**Pull request by mariusvniekerk**

### CODE REVIEW

**Comment by mariusvniekerk** [comment]
@conda-forge-admin , please rerender

**Comment by mariusvniekerk** [comment]
@msarahan

**Comment by mariusvniekerk** [comment]
@conda-forge-admin please rerender

**Comment by mariusvniekerk** [comment]
@conda-forge/help-c-cpp This seems to be failing due to a missing errno.h include under vc9 ?  Should we just exclude vc9 here?

**Comment by mariusvniekerk** [comment]
ping @mingwandroid @nehaljwani

**Comment by nehaljwani** [comment]
I am not sure what's happening here. I triggered a build for the recipe on our build system and I have attached the log. 

[krb5_vc9.txt](https://github.com/conda-forge/krb5-feedstock/files/2152572/krb5_vc9.txt)

**Comment by nehaljwani** [comment]
I ran the build on two machines, Win10 and Win7, with SDK v7.1 and SDK v6.0A respectively and both were successful.

**Comment by mingwandroid** [comment]
> cl    -I.\..\..\include -I.\..\..\include\krb5 -I..\..\windows\include -DWSHELPER=1 -DKRB5_DNS_LOOKUP=1 -DWIN32_LEAN_AND_MEAN -DKRB5_DEPRECATED=1 -DKRB5_PRIVATE -D_CRT_SECURE_NO_DEPRECATE -DUSE_LEASH=1  /Os /MD -nologo /EHsc /W3 -Fdobj\AMD64\rel\\ -FD  -Foobj\AMD64\rel\\ -c ...
threads.c

vs:

> cl    -I.\..\..\include -I.\..\..\include\krb5 -I..\..\windows\include -DWSHELPER=1 -DKRB5_DNS_LOOKUP=1 -DWIN32_LEAN_AND_MEAN -DKRB5_DEPRECATED=1 -DKRB5_PRIVATE -D_CRT_SECURE_NO_DEPRECATE -DUSE_LEASH=1  /Os /MD -nologo /EHsc /W3 -Fdobj\AMD64\rel\\ -FD  -Foobj\AMD64\rel\\ -c ...

ok nothing there .. comparing logs this is missing a patch:

```
Extracting download
Applying patch: 'C:\\ci\\containers\\0000388cert\\tmp\\build\\80754af9\\rsync-recipes\\krb5-1.16.1-c_compiler_vs2008-vc_9-cxx_compiler_vs2008-on-winbuilder\\eafnosupport.patch'
Trying to apply patch as-is
patching file 'src\include\port-sockets.h'
patching file 'src\include\k5-json.h'
patching file src/include/afxres.h
patching file 'src\windows\leashdll\lshutil.cpp'
```

**Comment by nehaljwani** [comment]
I think we need to fix the selectors anyway. They should all be:
```
# [win and vc<14]
```
Instead of:
```
# [win and py<34]
```
or 
```
# [win and py<35]
```

**Comment by nehaljwani** [comment]
I opened another PR at https://github.com/conda-forge/krb5-feedstock/pull/10 just to see what would happen if I fixed the selectors. The build proceeds fine. But during the tests...
```
## Package Plan ##
  environment location: C:\bld\krb5_1530442935081\_test_env
The following NEW packages will be INSTALLED:
    krb5:           1.16.1-hb4d044e_6 local      
    python:         3.6.5-1           conda-forge
    vc:             9-0               conda-forge
    vs2008_runtime: 9.0.30729.6161-0  conda-forge
    vs2015_runtime: 14.0.25420-0      conda-forge
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
(base) C:\bld\krb5_1530442935081\test_tmp>call "C:\Miniconda36-x64\Scripts\activate.bat" "C:\bld\krb5_1530442935081\_test_env" 
(C:\bld\krb5_1530442935081\_test_env) C:\bld\krb5_1530442935081\test_tmp>IF 0 NEQ 0 exit 1 
(C:\bld\krb5_1530442935081\_test_env) C:\bld\krb5_1530442935081\test_tmp>"C:\bld\krb5_1530442935081\_test_env\python.exe" -s "C:\bld\krb5_1530442935081\test_tmp\run_test.py" 
===== testing package: krb5-1.16.1-hb4d044e_6 =====
running run_test.py
```
Python 3 is being installed and it gets stuck while loading the DLL. I suspect that this is happening because of the use of vc features :(

**Comment by mariusvniekerk** [comment]
@nehaljwani, @msarahan This is a gross hack but gets us to a point where the package does buiild and function correctly for vc9 under windows

**Comment by mariusvniekerk** [comment]
osx is a flake

**Comment by mariusvniekerk** [comment]
ready for review @ocefpaf

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,42 +1,47 @@
...
-  skip: True  # [win and py>35]
-  features:
-    - vc9  # [win and py27]
-    - vc10  # [win and py34]
-    - vc14  # [win and py>=35]
+  number: 6
+  run_exports:
+    # stable within minor revisions: https://abi-laboratory.pro/tracker/timeline/krb5/
+    - {{ pin_subpackage('krb5', max_pin='x.x') }}
+  missing_dso_whitelist:          # [osx]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: We need to fix the linter to accept this.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,42 +1,47 @@
...
+    - lshutil.patch  # [win and vc<14]
 
 build:
-  number: 0
-  skip: True  # [win and py>35]
-  features:
-    - vc9  # [win and py27]
-    - vc10  # [win and py34]
-    - vc14  # [win and py>=35]
+  number: 6
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Can you change this to `1`? Our last build was `0`, right?

**Comment by ocefpaf** [comment]
LGTM I would like only to fix the build number before merging.

File: recipe/meta.yaml (hunk 1 above)
Comment (nehaljwani) [reply recipe/meta.yaml]: Fix is here: https://github.com/conda/conda-build/pull/2994

Comment (ocefpaf) [reply recipe/meta.yaml]: Thanks @nehaljwani!

File: recipe/meta.yaml (hunk 2 above)
Comment (mariusvniekerk) [reply recipe/meta.yaml]: This is also a higher version of krb5, so i'm going to set it to 0

Comment (ocefpaf) [reply recipe/meta.yaml]: Indeed. Thanks!

**Comment by ocefpaf** [comment]
We already know the tests are passing. Thanks a lot @mariusvniekerk! This one was not easy!!

**Pull request merged**